migrations/2025-07-22-091706_multiple_emails/up.sql

```diff
@@ -0,0 +1,102 @@
+-- Drop the unique constraint on user_id to allow multiple emails per user
+ALTER TABLE emails DROP CONSTRAINT emails_user_id_key;
+
+-- Limit users to 32 emails maximum
+CREATE FUNCTION enforce_max_emails_per_user()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF (SELECT COUNT(*) FROM emails WHERE user_id = NEW.user_id) > 32 THEN
+    RAISE EXCEPTION 'User cannot have more than 32 emails';
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER trigger_enforce_max_emails_per_user
+BEFORE INSERT ON emails
+FOR EACH ROW
+EXECUTE FUNCTION enforce_max_emails_per_user();
+
+-- Add a unique constraint for the combination of user_id and email
+ALTER TABLE emails ADD CONSTRAINT unique_user_email UNIQUE (user_id, email);
+
+-- Add a new column for identifying the primary email
+ALTER TABLE emails ADD COLUMN is_primary BOOLEAN DEFAULT FALSE NOT NULL;
+comment on column emails.is_primary is 'Whether this email is the primary email address for the user.';
+
+-- Set `is_primary` to true for existing emails
+UPDATE emails SET is_primary = true;
```

Comment:
we currently have 72k rows in the `emails` table. I'm a bit scared about running this as part of the migration, since the running migration will block the API server from restarting.

@eth3lbert any thoughts on this?

I guess we could extract the `add column is_primary` above to a dedicated migration/PR, deploy that to run the migration, then manually run the `update is_primary = true` without blocking the API server, and then merge/deploy the rest of this migration 🤔

Reply:
> I guess we could extract the `add column is_primary` above to a dedicated migration/PR, deploy that to run the migration, then manually run the `update is_primary = true` without blocking the API server, and then merge/deploy the rest of this migration 🤔

Yeah, I second this! I think this would be the most proper way to minimize potential issues caused by the introduced database migration.